cyrw/ioctl.go

```diff
@@ -276,15 +346,39 @@ func (d *Device) pollForIoctl(wantIoctlID uint16, buf []uint32) ([]byte, error)
 			return buf8[offset : offset+plen], err
 		}
 	}
-	return nil, errors.New("timeout")
+	return nil, errors.New("pollForIoctl timeout")
+}
+
+// waitForCredit waits for a credit to use for the next transaction
+func (d *Device) waitForCredit(buf []uint32) error {
+	if d.has_credit() {
+		return nil
+	}
+	for retries := 0; retries < 10; retries++ {
+		avail, length := d.f2PacketAvail()
```

Comment:
This block or two of code gets repeated in several functions, in slightly diff ways.  Can it be refactored into one func?

Reply:
It seems to me like we could refactor these two functions (pollForIoctl and waitForCredit) into a single composed function. They seem to be doing the same thing- and what's more: shouldn't pollForCtl should be checking the credit before wlan_read'ing?